src/overlays/actors/ovl_Obj_Smork/z_obj_smork.c

```diff
@@ -28,20 +28,125 @@ const ActorInit Obj_Smork_InitVars = {
+        if ((hours == 19) && (arg0 < minutes)) {
+            return 0;
+        }
+        ret = 3;
+    } else if (hours >= 11) {
+        if ((hours == 11) && (minutes < arg0)) {
+            return 0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Smork/ObjSmork_Init.s")
+        if (hours >= 14) {
+            return 0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Smork/ObjSmork_Destroy.s")
+        if ((hours == 13) && (arg0 < minutes)) {
+            return 0;
+        }
+        ret = 2;
+    } else if (hours >= 6) {
+        if ((hours == 6) && (minutes < arg0)) {
+            return 0;
+        }
+        if (hours >= 9) {
+            return 0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Smork/ObjSmork_Update.s")
+        if ((hours == 8) && (arg0 < minutes)) {
+            return 0;
+        }
+        ret = 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Smork/ObjSmork_Draw.s")
+    return ret;
+}
+
+void func_80A3D940(ObjSmork* this) {
+    u8 temp_v0 = func_80A3D680(this->unk_1C6);
+
+    if (temp_v0 != this->unk_1C4) {
+        this->unk_1C4 = temp_v0;
+        this->unk_1C6 = Rand_S16Offset(0, 59);
+    }
+
+    Math_ApproachF(&this->unk_1B8, (this->unk_1C4 == 0) ? 0.0f : 1.0f, 0.02f, 1000.0f);
+}
+
+void func_80A3D9C4(ObjSmork* this, GlobalContext* globalCtx) {
+    u8 sp57;
+    u8 sp56;
+    s32 i;
+
+    this->unk_1BC += 1.8f;
+    this->unk_1C0 += 0.6f;
+
+    sp57 = 127 - (u8)this->unk_1BC;
+    sp56 = 127 - (u8)this->unk_1C0;
+
+    this->unk_1B8 = CLAMP(this->unk_1B8, 0.0f, 1.0f);
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_148); i++) {
+        this->unk_148[i].v.cn[3] = ovl_Obj_Smork_Vtx_000C10[i].v.cn[3] * this->unk_1B8;
+    }
+
+    if (this->unk_1B8 > 0.0f) {
+        Matrix_InsertTranslation(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                                 MTXMODE_NEW);
+        this->actor.shape.rot.y = BINANG_ROT180(Camera_GetCamDirYaw(GET_ACTIVE_CAM(globalCtx)));
+        Matrix_RotateY(this->actor.shape.rot.y, MTXMODE_APPLY);
+        Matrix_Scale(0.1f, 0.1f, 0.0f, MTXMODE_APPLY);
+
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+
+        func_8012C28C(globalCtx->state.gfxCtx);
+
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, sp57, 0x20, 0x20, 1, 0, sp56, 0x20, 0x20));
+        gSPSegment(POLY_XLU_DISP++, 0x09, Lib_SegmentedToVirtual(this->unk_148));
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, object_f53_obj_DL_001C00);
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
+    }
+}
+
+void ObjSmork_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjSmork* this = THIS;
+
+    Lib_MemCpy(this->unk_148, ovl_Obj_Smork_Vtx_000C10, sizeof(ovl_Obj_Smork_Vtx_000C10));
```

Comment:
I personally prefer to use `sizeof(type)` instead of `sizeof(variable)`
```suggestion
    Lib_MemCpy(this->unk_148, ovl_Obj_Smork_Vtx_000C10, sizeof(Vtx) * ARRAY_COUNT(ovl_Obj_Smork_Vtx_000C10));
```